modules/lang-painless/src/test/java/org/elasticsearch/painless/RegexTests.java

```diff
@@ -125,6 +126,51 @@ public void testNamedGroup() {
         assertEquals("o", exec("Matcher m = /(?<first>f)(?<second>o)o/.matcher('foo'); m.find(); return m.namedGroup('second')"));
     }
 
+    public void testMultilineFlag() {
+        assertEquals(false, exec("'a\nb\nc' =~ /^b$/"));
+        assertEquals(true, exec("'a\nb\nc' =~ /^b$/m"));
+    }
```

Comment:
I think the most direct tests are preferable for unit tests. If we want them in the docs as examples lets put them there and have them tested there instead.

As far as Pattern, it would be really beneficial. whitelisting it should improve things as it would allow stuff like:  
````
for (def foo : /foo/.split(xxx)) {
  ...
``` 
as well as hooks into function stuff like Predicate/Streams.